client/validate.go

```diff
@@ -0,0 +1,87 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"golang.org/x/xerrors"
+)
+
+// ValidateApplyOptions carries options for ApplyValidationSet.
+type ValidateApplyOptions struct {
+	Flag  string
+	PinAt int
+}
+
+// ValidationSetResult holds information about a single validation set.
+type ValidationSetResult struct {
+	ValidationSet string   `json:"validation-set,omitempty"`
+	Mode          string   `json:"mode"`
+	Seq           int      `json:"seq,omitempty"`
+	Valid         bool     `json:"valid"`
+	Notes         []string `json:"notes,omitempty"`
+}
+
+// ApplyValidationSet applies or forgets the given validation set identified by account and name.
+func (client *Client) ApplyValidationSet(account, name string, opts *ValidateApplyOptions) error {
+	q := url.Values{}
+	q.Set("validation-set", fmt.Sprintf("%s/%s", account, name))
+	var postData struct {
+		Flag  string `json:"flag"`
+		PinAt string `json:"pin-at,omitempty"`
+	}
+	postData.Flag = opts.Flag
+	if opts.PinAt != 0 {
+		postData.PinAt = fmt.Sprintf("%d", opts.PinAt)
```

Comment:
Nitpick, would using an `int` in `postData` above provide the same result without any extra code? `PinAt` is defined as a string, which is somewhat odd for a number, can you at least add a comment why a different type is needed?

Reply:
Done.